src/pointfree/init.js

```diff
@@ -0,0 +1,25 @@
+/** @license ISC License (c) copyright 2016 original and current authors */
+/** @author Ian Hofmann-Hicks (evil) */
+
+const isFunction = require('../core/isFunction')
+const isNil = require('../core/isNil')
+
+const { Nothing, Just } = require('../core/Maybe')
+
+function init(m) {
+  if(!isNil(m)) {
+    if(isFunction(m.init)) {
+      return m.init()
+    }
+
+    if(isFunction(m.slice)) {
+      return m.length < 2
+        ? Nothing()
+        : Just(m.slice(0, -1))
+    }
+  }
+
+  throw new TypeError('init: Array, String, or List required')
```

Comment:
please see #405  WRT to the preferred voice for errors

```suggestion
  throw new TypeError('init: Argument must be an Array, String, or List')
```